### DISCUSSION TITLE
MoBi- Adding compartment to the model and connecting it to the molecule

### DISCUSSION THREAD

**Discussion by Sanjana2697** [Open-Systems-Pharmacology/Forum#936] [Questions & Problems]
Hi everyone,

I am trying to build a model in MoBi with an additional bone marrow compartment.
So, to build the model I have first imported the simulation from PK-Sim, then in the organism building block I tried to add an additional compartment of bone marrow. In bone marrow, added different physical compartments like plasma, blood cells, adipocytes, interstitial and intracellular. In the diagram connected all the physical compartments and also bone marrow to venous blood.
After this, in the reaction building block , tried to specify the bone marrow compartment in enzymes using the "Match all tag condition" in container criteria. 
Still after specifying all the parameters in bone marrow compartment and connecting the links between compartments , I am not able to run the simulation. 
On simulating, the errors like path from molecule to different compartments like venous blood, brain cannot be found. I am not able to catch the correct parameter which is leading to this error.
If anyone can help me find a correct way to build this model and which parameters will help me to connect the molecule to the compartment that would be helpful.
If you need any information on this to better understand the model,do let me know!
I just want to know how can the model takes in the new bone marrow compartment for the simulation with the necessary passive diffusion, enzymes and transporters already specified.
As I am new to building a drug model in MoBi, so any possible help will be very useful. Do let me know if you have any information.

Thank you

**Comment by wilbertdew** [comment]
Hi @Sanjana2697,

As with the creation of a tumor compartment, it would be most likely easier to save a reference organ as .pkml, import that organ as a new organ with a new name and then make all the necessary connections. There are more details in other discussions, such as [this one](https://github.com/Open-Systems-Pharmacology/Forum/issues/46). And [this one](https://github.com/Open-Systems-Pharmacology/MoBi/issues/1). Hope this helps a bit!
Best,

Wilbert

**Reply by Sanjana2697** [reply]
Hi @wilbertdew,

Thank you so much for the information. It was very helpful. I was able to solve passive diffusion errors, but still, I have few errors which I am not able to solve in the reaction building block.
The errors state that the partition coefficient in the bone marrow compartment for DNR is not found but I have added the parameter in the bone marrow compartment. Can you please guide me.
 
![Capture](https://user-images.githubusercontent.com/69060792/126225264-fa8cc5b3-cf90-4c58-99fe-0b2ce332078a.PNG)

Thank you
Regards,
Sanjana

**Reply by tztz117** [reply]
Hi Sanjana,

I met the similar issue as you when adding an tissue, I am wondering if you have solved your problem and how. thanks!
